ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -54,10 +54,31 @@ func (scbuilder *SiteConfigBuilder) Build(siteConfigTemp SiteConfig) (map[string
 		return clustersCRs, err
 	}
 
+	var clusterCRs []interface{}
 	for id, cluster := range siteConfigTemp.Spec.Clusters {
-		clusterCRs, err := scbuilder.getClusterCRs(id, siteConfigTemp)
-		if err != nil {
-			return clustersCRs, err
+		if cluster.ExtraManifestOnly {
+			// Generate extra manifests only
+			dataMap := make(map[string]interface{})
+			dataMap, err = scbuilder.getExtraManifest(dataMap, cluster)
+			if err != nil {
+				// Will return and fail if the end user extra-manifest having issues.
+				log.Printf("Error could not create extra-manifest %s.%s %s\n", cluster.ClusterName, cluster.ExtraManifestPath, err)
+				return clustersCRs, err
+			}
+
+			for _, manifest := range dataMap {
+				dataResult := make(map[string]interface{})
+				err := yaml.Unmarshal([]byte(manifest.(string)), &dataResult)
+				if err != nil {
+					log.Printf("Error: could not unmarshal string:(%s) (%s)\n", manifest.(string), err)
+				}
+				clusterCRs = append(clusterCRs, dataResult)
+			}
+		} else {
+			clusterCRs, err = scbuilder.getClusterCRs(id, siteConfigTemp)
+			if err != nil {
+				return clustersCRs, err
+			}
```

Comment:
I'd like to think about refactoring this a bit - If the `scbuilder.getClusterCRs(...)` function took the dataMap of extra manifests as input to the call, this logic (and the logic in the other calls) would be a lot cleaner.

I don't think we necessarily need to do it in this PR, but what do you thank of that approach in general?

Reply:
Updated